packages/stream/solana/StreamClient.ts

```diff
@@ -947,25 +947,45 @@ export default class SolanaStreamClient extends BaseStreamClient {
   }
 
   /**
-   * Utility function that checks whether associated token account for the recipient exists and adds an instruction to add if not
+   * Utility function that checks whether associated token accounts still exist and adds instructions to add them if not
    */
-  private async checkAssociatedTokenAccount(
-    data: CheckAssociatedTokenAccountData,
-    { invoker }: IInteractStreamSolanaExt,
+  private async checkAssociatedTokenAccounts(
+    data: CheckAssociatedTokenAccountsData,
+    { invoker, checkTokenAccounts }: IInteractStreamSolanaExt,
     ixs: TransactionInstruction[]
   ) {
-    const accountExists = await this.connection.getAccountInfo(data.recipientTokens);
-    if (!accountExists?.data) {
-      ixs.push(
-        createAssociatedTokenAccountInstruction(
-          invoker.publicKey!,
-          data.recipientTokens,
-          data.recipient,
-          data.mint,
-          TOKEN_PROGRAM_ID,
-          ASSOCIATED_TOKEN_PROGRAM_ID
-        )
-      );
+    if (!checkTokenAccounts) {
+      return;
+    }
+    const checkedKeys: Set<PublicKey> = new Set();
+    const accountArrays = [
+      [data.sender, data.senderTokens],
+      [data.recipient, data.recipientTokens],
+      [data.partner, data.partnerTokens],
+      [data.streamflowTreasury, data.streamflowTreasuryTokens],
+    ].filter((value) => {
+      if (checkedKeys.has(value[1])) {
+        return false;
+      }
+      checkedKeys.add(value[1]);
+      return true;
+    });
+    const response = await this.connection.getMultipleAccountsInfo(
+      accountArrays.map((item) => item[1])
+    );
+    for (let i = 0; i < response.length; i++) {
```

Comment:
Can we iterate here over accountArrays in forof loop?

for accountArray of accountArrays ...

Reply:
More clean then traditional for loop with indexes :)